### PR TITLE
Fix/structural response for analysis stage

### PR DESCRIPTION
**Pull request by armingeiser**
Quickfixes the response function and shape optimization tests for the new analysis stage.

@dbaumgaertner  we should think about modifiing the shape app in order to use Model instead of ModelPart on the top level.

@philbucher can you please have a look if the .json files look right? (the relative names of the submodelparts)

Note: The tests requiring mesh motion have not been tested, because i was not able to compile it.

### CODE REVIEW

**Comment by armingeiser** [comment]
@dbaumgaertner there is something strange with the solid examples --> Hook. 
We need to have a look on monday.

EDIT: solved, i accidentaly changed to CG iterative solver which gave rubbish results...

File: applications/StructuralMechanicsApplication/python_scripts/structural_response.py [hunk 1]
```diff
@@ -67,26 +68,32 @@ def __init__(self, identifier, response_settings, model_part = None):
...
+        self.primal_analysis = structural_mechanics_analysis.StructuralMechanicsAnalysis(model, ProjectParametersPrimal)
+        self.primal_model_part.AddNodalSolutionStepVariable(SHAPE_SENSITIVITY)
 
     def Initialize(self):
         self.primal_analysis.Initialize()
         self.response_function_utility.Initialize()
 
     def InitializeSolutionStep(self):
-        self.primal_analysis.InitializeTimeStep()
+        self.primal_analysis.time = self.primal_analysis.solver.AdvanceInTime(self.primal_analysis.time)
```
Comment (armingeiser) [inline applications/StructuralMechanicsApplication/python_scripts/structural_response.py:83]: This seems hacky...

**Comment by philbucher** [comment]
I think some more (hopefully minor) updates will be necessary after #2068 , I will take a look tmr

File: applications/StructuralMechanicsApplication/python_scripts/structural_response.py (hunk 1 above)
Comment (philbucher) [reply applications/StructuralMechanicsApplication/python_scripts/structural_response.py:83]: We have to speak abt how to integrate the StructuralAnalysis properly, riccardo doesn't want us to use it in this way, I will explain to you in person

**Comment by armingeiser** [comment]
#2068 did just reveal an error in one of the tests. Its fixed now.

**Review by philbucher — APPROVED** [review]
Looks ok to me, thx for fixing

**Pull request merged**